sshd-core/src/main/java/org/apache/sshd/server/forward/TcpipServerChannel.java

```diff
@@ -320,8 +322,14 @@ protected Closeable getInnerCloseable() {
                     @Override
                     @SuppressWarnings("synthetic-access")
                     protected CloseFuture doCloseGracefully() {
-                        executor.submit(() -> connector.close(false));
-                        return null;
+                        Future<CloseFuture> closeFeature = executor.submit(() -> connector.close(false));
+
+                        try {
+                            return closeFeature.get();
+                        } catch (InterruptedException | ExecutionException e) {
+                            log.error("CloseFuture execution completed with error ", e);
+                            return null;
```

Comment:
Does not seem to honor the **asynchronous** contract of this method - the whole idea of the `submit` call is **not** to wait for the connector to close. This is in contrast with `doCloseImmediately` which is **synchronous**.